metagraph/tests/annotation/test_annotated_dbg_helpers.cpp

```diff
@@ -1,71 +1,228 @@
-        );
+            );
+        } else if constexpr(!std::is_same_v<Annotation, RowDiffColumnAnnotator>) {
+            static_assert("Coordinates only supported for ColumnCompressed and RowDiff");
+        }
+    } else {
+        for (size_t i = 0; i < sequences.size(); ++i) {
+            anno_graph->annotate_sequence(std::string(sequences[i]), { labels[i] });
+        }
+
+        if constexpr(std::is_same_v<Annotation, ColumnCompressed<>>)
+            return anno_graph;
+
+        if constexpr(!std::is_same_v<Annotation, RowDiffColumnAnnotator>
+                && !std::is_same_v<Annotation, RowDiffCoordAnnotator>) {
+            return std::make_unique<AnnotatedDBG>(
+                graph,
+                std::unique_ptr<AnnotatedDBG::Annotator>(
+                    convert<Annotation>(
+                        std::move(dynamic_cast<ColumnCompressed<>&>(*anno_graph->annotator_))
+                    )
+                )
+            );
+        }
+    }
 
-    return anno_graph;
+    if constexpr(std::is_same_v<Annotation, RowDiffColumnAnnotator>) {
+        static_assert(std::is_same_v<Graph, DBGSuccinct>);
+        assert(dynamic_cast<const DBGSuccinct*>(base_graph.get()));
+        assert(!dynamic_cast<const DBGSuccinct*>(base_graph.get())->get_mask());
+
+        size_t num_columns = anno_graph->get_annotator().get_label_encoder().size();
+        std::filesystem::path tmp_dir = utils::create_temp_dir("", "test_col");
+        std::string out_path = tmp_dir/"test_col";
+        anno_graph->get_annotator().serialize(out_path);
+        std::vector<std::string> files { out_path + ColumnCompressed<>::kExtension };
+        base_graph->serialize(out_path);
+        std::string graph_fname = out_path + base_graph->file_extension();
+        if (!std::filesystem::exists(files[0])) {
+            logger->error("Failed to serialize annotation to {}.", files[0]);
+            std::exit(1);
+        }
+        if (!std::filesystem::exists(graph_fname)) {
+            logger->error("Failed to serialize graph to {}.", graph_fname);
+            std::exit(1);
+        }
+
+        {
+            std::filesystem::path swap_dir = utils::create_temp_dir("", "swap_col");
+            convert_to_row_diff(files, graph_fname, 100000000, 5, tmp_dir, swap_dir,
+                                static_cast<RowDiffStage>(0), out_path + ".row_count", false, coordinates);
+            convert_to_row_diff(files, graph_fname, 100000000, 5, tmp_dir, swap_dir,
+                                static_cast<RowDiffStage>(1), out_path + ".row_reduction", false, coordinates);
+            convert_to_row_diff(files, graph_fname, 100000000, 5, tmp_dir, swap_dir,
+                                static_cast<RowDiffStage>(2), out_path, false, coordinates);
+        }
+
+        auto annotator = std::make_unique<ColumnCompressed<>>(0);
+        if (!annotator->merge_load(files)) {
+            logger->error("Cannot load annotations");
+            exit(1);
+        }
```

Comment:
So you converted it to row-diff but then load again the same untransformed columns... Am I missing something?

Reply:
I'm only using this for loading. Below, I'm wrapping the underlying columns in a RowDiff matrix.

Reply:
So you wrap the original untransformed columns as row-diff?

Reply:
For plain row-diff without coordinates, it writes to `.row_diff` columns. So you can't load them as `ColumnCompressed<>`. See how it's done in the main converter.

Reply:
Ahh, I see, that's confusing... I had previously done it this way for annotations with coordinates, so I thought it would also work the same way without coordinates.

Reply:
That sounds like it should fix the issue, thanks!